application/src/main/java/org/opentripplanner/updater/trip/gtfs/TripTimesUpdater.java

```diff
@@ -93,17 +96,28 @@ public static Result<TripTimesPatch, UpdateError> createUpdatedTripTimesFromGTFS
     }
 
     RealTimeTripTimesBuilder builder = tripTimes.createRealTimeWithoutScheduledTimes();
-    List<Integer> skippedStopIndices = new ArrayList<>();
+    if (tripUpdate.hasTripProperties()) {
+      var tripProperties = tripUpdate.getTripProperties();
+      if (tripProperties.hasTripHeadsign()) {
+        builder.withTripHeadsign(I18NString.of(tripProperties.getTripHeadsign()));
+      }
+      // TODO: add support for changing trip short name
+    }
+
```

Comment:
I think we should introduce a wrapper object like `AddedStopTime` where we can put all of this very verbose getter logic and return `Optional<I18NString>`.

Then it become easy to do a `wrapper.tripHeadsign().ifPresent(h -> builder.withTripHeadsign(h))`.